velox/expression/CastExpr.cpp

```diff
@@ -520,63 +520,80 @@ VectorPtr CastExpr::applyDecimal(
   return castResult;
 }
 
-/// Apply casting between a custom type and another type.
-/// @param castTo The boolean indicating whether to cast an input to the custom
-/// type or from the custom type
-/// @param input The input vector
-/// @param allRows The selectivity vector of all rows in input
-/// @param nonNullRows The selectivity vector of non-null rows in input
-/// @param castOperator The cast operator for the custom type
-/// @param thisType The custom type
-/// @param otherType The other type involved in this casting
-/// @param context The context
-/// @param nullOnFailure Whether this is a cast or try_cast operation
-/// @param result The output vector
-template <bool castTo>
-void applyCustomTypeCast(
-    VectorPtr& input,
-    DecodedVector& inputDecoded,
-    const SelectivityVector& allRows,
-    const SelectivityVector& nonNullRows,
-    const CastOperatorPtr& castOperator,
-    const TypePtr& thisType,
-    const TypePtr& otherType,
+/// Apply casting from a non-custom type to a complex type.
+/// @param decoded The decoded input vector
+/// @param rows The selectivity vector of non-null rows in decoded.base().
+void CastExpr::applyComplexType(
```

Comment:
Usually, when we pass 'rows' and 'decodedVector', 'rows' are top-level rows that can be used with decodedVector.isNullAt() and valueAt. Here, 'rows' are row numbers of the base vector. I wonder if it would be clearer to pass the base vector itself to this method instead of DecodedVector. I assume the logic here only ever uses decoded.base() and nothing else.